### PR TITLE
Np 42020 create published as external

### PR DESCRIPTION
**Pull request by sondrev**

### CODE REVIEW

File: publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java [hunk 1]
```diff
@@ -76,6 +76,7 @@ public class ResourceService extends ServiceWithTransactions {
     public static final String RESOURCE_CANNOT_BE_DELETED_ERROR_MESSAGE = "Resource cannot be deleted: ";
     public static final int MAX_SIZE_OF_BATCH_REQUEST = 20;
     private static final Logger logger = LoggerFactory.getLogger(ResourceService.class);
+    public static final String NOT_PUBLISHABLE = "Publication is not publiashable. Check main title and doi";
```
Comment (sondrev) [inline publication-commons/src/main/java/no/unit/nva/publication/service/impl/ResourceService.java:79]: typo in publiashable

File: publication-commons/src/main/java/no/unit/nva/publication/RequestUtil.java [hunk 2]
```diff
@@ -65,7 +65,7 @@ public static UserInstance createExternalUserInstance(RequestInfo requestInfo,
             client.getCristinUrgUri()
         );
 
-        return UserInstance.create(resourceOwner, client.getCustomerUri());
+        return UserInstance.createMachineUser(resourceOwner, client.getCustomerUri());
```
Comment (sondrev) [inline publication-commons/src/main/java/no/unit/nva/publication/RequestUtil.java:68]: machineUser->externalClient
Forgot to refactor this

File: publication-rest/src/test/java/no/unit/nva/publication/create/CreatePublicationHandlerTest.java [hunk 3]
```diff
@@ -133,6 +139,51 @@ void requestToHandlerReturnsMinRequiredFieldsWhenRequestContainsEmptyResource()
         assertExistenceOfMinimumRequiredFields(publicationResponse);
     }
 
+    @Test
+    void requestToHandlerWithStatusPublishedFromAnRegularUserShouldPersistDocumentAsDraft() throws Exception {
+        var request = createEmptyPublicationRequest();
+        request.setStatus(PublicationStatus.PUBLISHED);
+        InputStream inputStream = createPublicationRequest(request);
```
Comment (brinxmat) [inline publication-rest/src/test/java/no/unit/nva/publication/create/CreatePublicationHandlerTest.java:146]: var

File: publication-rest/src/test/java/no/unit/nva/publication/create/CreatePublicationHandlerTest.java [hunk 4]
```diff
@@ -133,6 +139,51 @@ void requestToHandlerReturnsMinRequiredFieldsWhenRequestContainsEmptyResource()
...
+    @ParameterizedTest(name = "requestToHandlerWithStatusFromAnExternalClientShouldPersistDocumentWithSameStatus with"
+                              + " status: \"{0}\"")
+    @ValueSource(strings = {"DRAFT", "PUBLISHED"})
+    void requestToHandlerWithStatusFromAnExternalClientShouldPersistDocumentWithSameStatus(String statusString)
+        throws Exception {
+        var status = PublicationStatus.valueOf(statusString);
+        var request = createEmptyPublicationRequest();
+        request.setStatus(status);
+        request.setEntityDescription(randomPublishableEntityDescription());
+        InputStream inputStream = requestFromExternalClient(request);
```
Comment (brinxmat) [inline publication-rest/src/test/java/no/unit/nva/publication/create/CreatePublicationHandlerTest.java:164]: var

File: publication-rest/src/test/java/no/unit/nva/publication/create/CreatePublicationHandlerTest.java [hunk 5]
```diff
@@ -133,6 +139,51 @@ void requestToHandlerReturnsMinRequiredFieldsWhenRequestContainsEmptyResource()
...
+        var publicationResponse = actual.getBodyObject(PublicationResponse.class);
+        assertThat(publicationResponse.getStatus(), is(equalTo(status)));
+    }
+
+    @Test
+    void shouldThrowBadRequestWhenAnExternalClientTriesToCreatePublishedDocumentWithoutTitleAndDoiRef()
+        throws Exception {
+        var request = createEmptyPublicationRequest();
+        request.setStatus(PublicationStatus.PUBLISHED);
+        InputStream inputStream = requestFromExternalClient(request);
```
Comment (brinxmat) [inline publication-rest/src/test/java/no/unit/nva/publication/create/CreatePublicationHandlerTest.java:178]: var

File: publication-rest/src/test/java/no/unit/nva/publication/create/CreatePublicationHandlerTest.java [hunk 6]
```diff
@@ -133,6 +139,51 @@ void requestToHandlerReturnsMinRequiredFieldsWhenRequestContainsEmptyResource()
         assertExistenceOfMinimumRequiredFields(publicationResponse);
     }
 
+    @Test
+    void requestToHandlerWithStatusPublishedFromAnRegularUserShouldPersistDocumentAsDraft() throws Exception {
```
Comment (brinxmat) [inline publication-rest/src/test/java/no/unit/nva/publication/create/CreatePublicationHandlerTest.java:143]: shouldPersistDraftWhenRegularUserAttemptsToPersistPublicationWithStatusPublished

File: publication-rest/src/test/java/no/unit/nva/publication/create/CreatePublicationHandlerTest.java [hunk 7]
```diff
@@ -133,6 +139,51 @@ void requestToHandlerReturnsMinRequiredFieldsWhenRequestContainsEmptyResource()
...
+        var actual = GatewayResponse.fromOutputStream(outputStream, PublicationResponse.class);
+        assertThat(actual.getStatusCode(), is(equalTo(HttpURLConnection.HTTP_CREATED)));
+        var publicationResponse = actual.getBodyObject(PublicationResponse.class);
+        assertThat(publicationResponse.getStatus(), is(equalTo(PublicationStatus.DRAFT)));
+    }
+
+    @ParameterizedTest(name = "requestToHandlerWithStatusFromAnExternalClientShouldPersistDocumentWithSameStatus with"
+                              + " status: \"{0}\"")
+    @ValueSource(strings = {"DRAFT", "PUBLISHED"})
+    void requestToHandlerWithStatusFromAnExternalClientShouldPersistDocumentWithSameStatus(String statusString)
```
Comment (brinxmat) [inline publication-rest/src/test/java/no/unit/nva/publication/create/CreatePublicationHandlerTest.java:158]: shouldPersistProvidedStatusWhenMachineUserPersistsPublicationWithAnyStatus

File: publication-rest/src/test/java/no/unit/nva/publication/delete/DeletePublicationHandlerTest.java [hunk 8]
```diff
@@ -166,7 +168,8 @@ void handleRequestReturnsErrorWhenCallerIsNotOwnerOfPublication() throws IOExcep
     }
 
     @Test
-    void handleRequestReturnsErrorWhenCallerIsNotOwnerOfPublicationAndCalledIsExternalClient() throws IOException {
+    void handleRequestReturnsErrorWhenCallerIsNotOwnerOfPublicationAndCalledIsExternalClient()
```
Comment (brinxmat) [inline publication-rest/src/test/java/no/unit/nva/publication/delete/DeletePublicationHandlerTest.java:171]: shouldReturn

File: publication-rest/src/test/java/no/unit/nva/publication/delete/DeletePublicationHandlerTest.java [hunk 9]
```diff
@@ -186,7 +189,8 @@ void handleRequestReturnsErrorWhenCallerIsNotOwnerOfPublicationAndCalledIsExtern
     }
 
     @Test
-    void handleRequestReturnsUnauthorizedWhenCallerIsMissingClientId() throws IOException, NotFoundException {
+    void handleRequestReturnsUnauthorizedWhenCallerIsMissingClientId()
```
Comment (brinxmat) [inline publication-rest/src/test/java/no/unit/nva/publication/delete/DeletePublicationHandlerTest.java:192]: shouldReturn

File: publication-rest/src/test/java/no/unit/nva/publication/update/UpdatePublicationHandlerTest.java [hunk 10]
```diff
@@ -128,7 +129,7 @@ void handlerUpdatesPublicationWhenInputIsValidAndUserIsResourceOwner()
     }
 
     @Test
-    void handlerUpdatesPublicationWhenInputIsValidAndUserIsExternalClient() throws IOException {
+    void handlerUpdatesPublicationWhenInputIsValidAndUserIsExternalClient() throws IOException, BadRequestException {
```
Comment (brinxmat) [inline publication-rest/src/test/java/no/unit/nva/publication/update/UpdatePublicationHandlerTest.java:132]: should…

File: publication-rest/src/test/java/no/unit/nva/publication/update/UpdatePublicationHandlerTest.java [hunk 11]
```diff
@@ -256,7 +257,8 @@ void handlerThrowsExceptionWhenInputIsValidUserHasRightToEditAnyResourceInOwnIns
     }
 
     @Test
-    void handlerThrowsNotFoundWhenExternalClientTriesToUpdateResourcesCreatedByOthers() throws IOException {
+    void handlerThrowsNotFoundWhenExternalClientTriesToUpdateResourcesCreatedByOthers()
```
Comment (brinxmat) [inline publication-rest/src/test/java/no/unit/nva/publication/update/UpdatePublicationHandlerTest.java:260]: should

**Review by brinxmat — CHANGES_REQUESTED** [review]

**Pull request merged**